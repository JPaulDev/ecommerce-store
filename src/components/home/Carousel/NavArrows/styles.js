import styled from 'styled-components';

export const List = styled.ul`
  display: flex;
  justify-content: space-between;
  list-style: none;
  padding: 0 15px;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  width: 100%;
  z-index: 10;
`;

export const Button = styled.button`
  align-items: center;
  background-color: white;
  border-radius: 50%;
  box-shadow: 1px 4px 4px rgb(0 0 0 / 50%);
  cursor: pointer;
  display: flex;
  height: 32px;
  justify-content: center;
  user-select: none;
  width: 32px;
`;
