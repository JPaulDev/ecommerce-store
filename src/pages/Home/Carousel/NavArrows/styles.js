import styled from 'styled-components';

export const PrevSlideBtn = styled.button`
  background-color: white;
  border-radius: 50%;
  box-shadow: 1px 4px 4px rgb(0 0 0 / 50%);
  cursor: pointer;
  display: flex;
  left: 15px;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  user-select: none;
  z-index: 10;
`;

export const NextSlideBtn = styled(PrevSlideBtn)`
  left: initial;
  right: 15px;
`;
