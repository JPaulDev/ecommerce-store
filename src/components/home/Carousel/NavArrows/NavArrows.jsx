import ChevronLeft from '../../../icons/ChevronLeft';
import ChevronRight from '../../../icons/ChevronRight';
import * as Styled from './styles';

export default function NavArrows({ onChangeSlide }) {
  return (
    <Styled.List>
      <li>
        <Styled.Button
          type="button"
          title="Previous"
          aria-label="Previous Slide"
          onClick={() => onChangeSlide('PREV')}
        >
          <ChevronLeft width={14} height={19} />
        </Styled.Button>
      </li>
      <li>
        <Styled.Button
          type="button"
          title="Next"
          aria-label="Next Slide"
          onClick={() => onChangeSlide('NEXT')}
        >
          <ChevronRight width={14} height={19} />
        </Styled.Button>
      </li>
    </Styled.List>
  );
}
