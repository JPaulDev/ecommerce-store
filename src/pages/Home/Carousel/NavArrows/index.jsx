import MaterialIcon from '../../../../components/MaterialIcon';
import * as Styled from './styles';

export default function NavArrows({ onChangeSlide }) {
  return (
    <>
      <Styled.PrevSlideBtn
        type="button"
        title="Previous"
        aria-label="Previous Slide"
        onClick={() => onChangeSlide('PREV')}
      >
        <MaterialIcon iconSize="32px">chevron_left</MaterialIcon>
      </Styled.PrevSlideBtn>
      <Styled.NextSlideBtn
        type="button"
        title="Next"
        aria-label="Next Slide"
        onClick={() => onChangeSlide('NEXT')}
      >
        <MaterialIcon iconSize="32px">chevron_right</MaterialIcon>
      </Styled.NextSlideBtn>
    </>
  );
}
