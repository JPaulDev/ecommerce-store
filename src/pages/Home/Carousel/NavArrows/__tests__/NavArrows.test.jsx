import renderer from 'react-test-renderer';
import NavArrows from '../index';

describe('NavArrows component', () => {
  it('renders two buttons with left and right arrows', () => {
    const tree = renderer.create(<NavArrows />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
