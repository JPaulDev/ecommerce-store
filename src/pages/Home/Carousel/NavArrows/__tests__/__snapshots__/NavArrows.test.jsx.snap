// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavArrows component renders two buttons with left and right arrows 1`] = `
Array [
  <button
    aria-label="Previous Slide"
    className="sc-gsnTZi jxroVI"
    onClick={[Function]}
    title="Previous"
    type="button"
  >
    <i
      className="sc-bczRLJ efclUT material-icons"
    >
      chevron_left
    </i>
  </button>,
  <button
    aria-label="Next Slide"
    className="sc-gsnTZi sc-dkzDqf jxroVI gFkErQ"
    onClick={[Function]}
    title="Next"
    type="button"
  >
    <i
      className="sc-bczRLJ efclUT material-icons"
    >
      chevron_right
    </i>
  </button>,
]
`;
